### DISCUSSION TITLE
Dialog render not into body

### DISCUSSION THREAD

**Discussion by prbxr** [ariakit/ariakit#779] [Questions] [answered]
Our app is embedded into a website.
I want to global container media related classes to influence modals.
This is why I want to render them under the root element of the embedded app.
How to do it with Reakit?

**Comment by diegohaz** [comment, marked as answer]
Not sure if I understand your question. But you can try to use `PortalContext`:
```jsx
import { PortalContext } from "reakit/Portal";

<PortalContext.Provider value={anotherDOMElement}>
  <Dialog />
</PortalContext.Provider>
```

**Reply by prbxr** [reply]
Thank you for the fast response!

Great. That actually works. It solves the css sharing problem.
But turns out that my problem is bigger.

I would try to explain better the situation:
Our App is embedded in  a host website. Meaning, it is given a Div and in that Div it will live.
In that div it opens a Modal.

Now, the regular Modal behaves like:
1. Disables scroll on the Body
2. Closes when clicked outside wherever

THAT Modal should disable scroll ONLY inside that root Div.
And should close ONLY when clicked INSIDE that root Div.

It should live by itself and should not affect the host page.
It can't be embedded in an iframe.

You understand the situation?

**Reply by diegohaz** [reply]
Hmm! You can disable the default behavior by passing `false` to `hideOnClickOutside` and `preventBodyScroll` props and add your custom behavior.
```jsx
<Dialog hideOnClickOutside={false} preventBodyScroll={false} />
```
